src/network/hobbits/hobbitsUri.ts

```diff
@@ -0,0 +1,76 @@
+import PeerInfo from "peer-info";
+import {peerInfoToAddress} from "./util";
+import {promisify} from "util";
+import {add} from "winston";
+
+export interface IHobbitsUriOpts {
+  scheme?: string;
+  identity?: string;
+  host?: string;
+  port?: number;
+  uriString?: string;
+}
+
+export class HobbitsUri {
+  public scheme: string;
+  public identity: string;
+  public host: string;
+  public port: number;
+  public constructor(opts?: IHobbitsUriOpts) {
+    if(opts.uriString) {
+      // TODO: More validation on identity, host, port
+      let regx = /^hob\+(tcp|udp):\/\/(?:([0-9a-f]*)@)?([a-zA-Z0-9.]*):([0-9]+)$/g;
+      let match = regx.exec(opts.uriString);
+      if (match) {
+        this.scheme = match[1];
+        if (match[2]) {
+          this.identity = match[2];
+        }
+        this.host = match[3];
+        this.port = parseInt(match[4]);
+      }
```

Comment:
extract this to some function like validateHobbitsUri

Reply:
Like this? 
```
public validateHobbitsUri(uriString: string): boolean{
    // TODO: More validation on identity, host, port
    let regx = /^hob\+(tcp|udp):\/\/(?:([0-9a-f]*)@)?([a-zA-Z0-9.]*):([0-9]+)$/g;
    let match = regx.exec(uriString);
    if (match) {
      this.scheme = match[1];
      if (match[2]) {
        this.identity = match[2];
      }
      this.host = match[3];
      this.port = parseInt(match[4]);
      return true;
    }
    return false;
  }
```

Reply:
I would lean toward returning {scheme, identity, host, port} if ok and null otherwise